dandischema/metadata.py

```diff
@@ -158,16 +164,20 @@ def migrate(
         id = obj.get("id")
         if not id.startswith("DANDI:"):
             obj["id"] = f'DANDI:{obj["id"]}'
-        for contrib in obj["contributor"]:
+        for contrib in obj.get("contributor", []):
+            if not contrib.get("roleName"):
+                continue
             contrib["roleName"] = [
                 val.replace("dandi:", "dcite:") for val in contrib["roleName"]
             ]
             for affiliation in contrib.get("affiliation", []):
                 affiliation["schemaKey"] = "Affiliation"
-        for contrib in obj["relatedResource"]:
+        for contrib in obj.get("relatedResource", []):
             contrib["relation"] = contrib["relation"].replace("dandi:", "dcite:")
-        for access in obj["access"]:
+        for access in obj.get("access", []):
             access["status"] = "dandi:OpenAccess"
+        else:
+            obj["access"] = [{"status": "dandi:OpenAccess"}]
```

Comment:
this Python pattern of `for ... else` was always confusing to me since it is pretty much `for ... if_I_did_not_break` and not  `for ... if_I_had_nothing_to_loop_through`, and I am afraid you might have been confused by it as well since 
```
$> cat /tmp/1.py   
for i in [1]:
   print(i)
else:
   print("or else")

$> python3 /tmp/1.py
1
or else
```
so you would end up just overwriting what you looped through first.  So you do need an explicit `if` for absent `access` instead.

Reply:
indeed. thanks for that. i don't use it much, but i guess when i use it, i use it incorrectly!